code/dce-downloader/node/dce-common.js

```diff
@@ -0,0 +1,132 @@
+
+const request = require('request-promise');
+const crypto = require('crypto');
+const util = require('util');
+const pathLib = require('path');
+const fs = require('fs');
+
+module.exports = {
+    readConfig : function readConfig(config) {
+        try {
+            return JSON.parse(fs.readFileSync(config, 'utf8'));
+        } catch (e) {
+            console.error("Error reading %s: %s", config, e.toString());
+            process.exit(1);
+        }
+    },
+
+    getEnvironment : function getEnvironment(config, env) {
+        if (!(env in config)) {
+            console.error(util.format("Environment \"%s\" not found in config", env));
+            return undefined;
+        }
+
+        const environment = config[env];
+        if (!environment.url ||
+            !environment.passkey ||
+            !environment.secret) {
+
+            console.error(util.format("Environment \"%s\" is missing one or more properties: %s", env, JSON.stringify(environment)));
```

Comment:
Node supports string interpolation (syntax support for formatting). Use the backtick (`) syntax.

```javascript
console.error(`Environment ${env} is missing one or more properties: ${JSON.stringify(environment)}`);
```